### ISSUE TITLE
Request TypeScript Strict Mode Support

### ISSUE THREAD

**Issue by Luna-Runa** [tamimaj/nestjs-redis-streams#15]
Hi, I've been using the library 'nestjs-redis-streams' and it's very useful. However, I noticed in my project that when I enabled the 'strict' mode of TypeScript, I noticed several types of errors.

The cause of the problem appears to be that the library 'nestjs-redis-streams' does not correspond to 'strict' mode. For example, a type error occurs when some variables can be 'null' but the type does not explicitly address the possibility.

Examples include.
```typescript
// redisStreamClient.coreModule.ts
  public static createAsyncClientProvider(options: RedisStreamModuleAsyncOptions): Provider {
    if (!(options.useExisting || options.useFactory || options.useClass)) {
      throw new Error('Invalid configuration. Must provide useFactory, useClass or useExisting')
    }

    if (options.useFactory) {
      return {
        provide: REDIS_STREAM_CLIENT_MODULE_OPTIONS,
        useFactory: options.useFactory,
        inject: options.inject || [],
      }
    }

    return {
      provide: REDIS_STREAM_CLIENT_MODULE_OPTIONS,
      useFactory: async (optionsFactory: RedisStreamClientModuleOptionsFactory) =>
        optionsFactory.createRedisStreamClientModuleOptions(),
      inject: [options.useClass || options.useExisting], // this
    /* The type 'Type<RedisStreamClientModuleOptionsFactory> | undefined' cannot be assigned to the type 'InjectionToken | OptionalFactoryDependency'. 
        The type 'undefined' cannot be assigned to the type 'InjectionToken | OptionalFactoryDependency'.ts(2322) */
    }
  }

// interfaces.ts
export interface RedisStreamModuleAsyncOptions extends Pick<ModuleMetadata, 'imports'> {
  useExisting?: Type<RedisStreamClientModuleOptionsFactory>
  useClass?: Type<RedisStreamClientModuleOptionsFactory> // this
  useFactory?: (...args: any[]) => Promise<ClientConstructorOptions> | ClientConstructorOptions
  inject?: any[]
}
```

To address this issue, we propose to modify the type definition of the library to fit the 'strict' mode, which will make the library more reliable for projects using 'strict' mode.

I chose to manually modify the code of this library, but if official support is provided at the library level, it would be convenient for other users as well. Therefore, I would appreciate it if you could consider this issue.

Thank you.

**Comment by tamimaj** [comment]
Hey @Luna-Runa,thank you very much for spotting and reporting this issue. Do you want to open a pull request with the type fixes if you have them? Otherwise I can look at it ASAP.

**Comment by tamimaj** [comment]
> There is a customized part, but if you want, I can revise it a little and open the PR.

Yes please go ahead. I can review it later and publish a new version once everything is ok.

**Comment by tamimaj** [comment]
@Luna-Runa Thanks for your help. This was merged and deployed v1.0.5.
